libs/langchain/langchain/cache.py

```diff
@@ -470,21 +451,63 @@ def _get_gptcache(self, llm_string: str) -> Any:
             _gptcache = self._new_gptcache(llm_string)
         return _gptcache
 
-    def lookup(self, prompt: str, llm_string: str) -> Optional[RETURN_VAL_TYPE]:
+    def lookup(self, prompt: str, llm_string: str) -> Optional[List[ChatGeneration]]:
         """Look up the cache data.
-        First, retrieve the corresponding cache object using the `llm_string` parameter,
-        and then retrieve the data from the cache based on the `prompt`.
+
+        Args:
+            prompt (str): The prompt to search for.
+            llm_string (str): The language model string.
+
+        Returns:
+            Optional[List[ChatGeneration]]: \
+                List of ChatGeneration instances if found, else None.
         """
-        from gptcache.adapter.api import get
+        try:
+            from gptcache.adapter.api import get
+        except ImportError:
+            raise ImportError(
+                "Could not import gptcache python package. "
+                "Please install it with `pip install gptcache`."
+            )
+        gpt_cache = self.gptcache_dict.get(llm_string)
 
-        _gptcache = self._get_gptcache(llm_string)
+        if not gpt_cache:
+            return None
 
-        res = get(prompt, cache_obj=_gptcache)
-        if res:
-            return [
-                Generation(**generation_dict) for generation_dict in json.loads(res)
-            ]
-        return None
+        if isinstance(prompt, str) and prompt.startswith("["):
+            try:
+                prompt = json.loads(prompt)
+            except json.JSONDecodeError:
+                raise ValueError(
+                    "Prompt must be a valid JSON string or a list of JSON strings."
+                )
+
+        if isinstance(prompt, list):
+            prompt = prompt[0].get("kwargs", {}).get("content", "")
+
+        response = get(prompt, cache_obj=gpt_cache)
+
+        if not response:
+            return None
+
+        response_data = loads(response)
+
+        generations = [
+            ChatGeneration(
```

Comment:
@kdcokenny You have replaced Generation with ChatGeneration. Will this break the existing functionality? While making changes, please take into account that lookup function is used for all GPTCache use cases. Every change should be backward compatible.

Reply:
It has been a while since I've changed this, but it passes all of the previous GPTCache test cases as well as I manually tested ChatOpenAI and OpenAI LLMs and they worked fine.